--- mysql-5.5.31/vio/viossl.c.openssl	2013-04-21 18:45:05.786362942 +0200
+++ mysql-5.5.31/vio/viossl.c	2013-04-21 18:52:19.272402084 +0200
@@ -172,8 +172,10 @@
   SSL_SESSION_set_timeout(SSL_get_session(ssl), timeout);
   SSL_set_fd(ssl, vio->sd);
 #ifndef HAVE_YASSL
+#ifdef SSL_OP_NO_COMPRESSION
   SSL_set_options(ssl, SSL_OP_NO_COMPRESSION);
 #endif
+#endif
 
   if ((r= connect_accept_func(ssl)) < 1)
   {
